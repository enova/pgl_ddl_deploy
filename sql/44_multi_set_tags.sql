SET client_min_messages = warning;

SET ROLE test_pgl_ddl_deploy;

CREATE SCHEMA viewer;

--Should be handled by separate set_config
CREATE TABLE viewer.foo(id int primary key);

--Should be handled by separate set_config
CREATE VIEW viewer.vw_foo AS
SELECT * FROM viewer.foo;

SELECT c.create_tags, c.set_name, ddl_sql_raw, ddl_sql_sent
FROM pgl_ddl_deploy.events e
INNER JOIN pgl_ddl_deploy.set_configs c ON c.id = e.set_config_id
WHERE c.set_name = 'test1'
ORDER BY e.id DESC LIMIT 4;

DROP VIEW viewer.vw_foo;
DROP TABLE viewer.foo CASCADE;
DROP SCHEMA viewer;

SELECT c.drop_tags, c.set_name, ddl_sql_raw, ddl_sql_sent
FROM pgl_ddl_deploy.events e
INNER JOIN pgl_ddl_deploy.set_configs c ON c.id = e.set_config_id
WHERE c.set_name = 'test1'
ORDER BY e.id DESC LIMIT 4;

SELECT * FROM pgl_ddl_deploy.exceptions;

DO $$
DECLARE v_ct INT;
BEGIN

IF current_setting('server_version_num')::INT >= 100000 THEN
    SELECT COUNT(1) INTO v_ct FROM pg_publication_tables WHERE schemaname = 'pgl_ddl_deploy' AND tablename = 'queue';
    RAISE LOG 'v_ct: %', v_ct;
    PERFORM verify_count(v_ct, 8);
END IF;

END$$;
