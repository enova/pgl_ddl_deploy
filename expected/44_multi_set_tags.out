SET client_min_messages = warning;
SET ROLE test_pgl_ddl_deploy;
CREATE SCHEMA viewer;
--Should be handled by separate set_config
CREATE TABLE viewer.foo(id int primary key);
--Should be handled by separate set_config
CREATE VIEW viewer.vw_foo AS
SELECT * FROM viewer.foo;
SELECT c.create_tags, c.set_name, ddl_sql_raw, ddl_sql_sent
FROM pgl_ddl_deploy.events e
INNER JOIN pgl_ddl_deploy.set_configs c ON c.id = e.set_config_id
WHERE c.set_name = 'test1'
ORDER BY e.id DESC LIMIT 4;
                                                 create_tags                                                  | set_name |                 ddl_sql_raw                  |                 ddl_sql_sent                 
--------------------------------------------------------------------------------------------------------------+----------+----------------------------------------------+----------------------------------------------
 {"CREATE VIEW","ALTER VIEW","CREATE FUNCTION","ALTER FUNCTION"}                                              | test1    | CREATE VIEW viewer.vw_foo AS                +| CREATE VIEW viewer.vw_foo AS                +
                                                                                                              |          | SELECT * FROM viewer.foo;                    | SELECT * FROM viewer.foo;
 {"ALTER TABLE","CREATE SEQUENCE","ALTER SEQUENCE","CREATE SCHEMA","CREATE TABLE","CREATE TYPE","ALTER TYPE"} | test1    | CREATE TABLE viewer.foo(id int primary key); | CREATE TABLE viewer.foo(id int primary key);
 {"ALTER TABLE","CREATE SEQUENCE","ALTER SEQUENCE","CREATE SCHEMA","CREATE TABLE","CREATE TYPE","ALTER TYPE"} | test1    | CREATE SCHEMA viewer;                        | CREATE SCHEMA viewer;
 {"ALTER TABLE","CREATE SEQUENCE","ALTER SEQUENCE","CREATE SCHEMA","CREATE TABLE","CREATE TYPE","ALTER TYPE"} | test1    | DROP TABLE foobar.foo CASCADE;               | DROP TABLE foobar.foo CASCADE;
(4 rows)

DROP VIEW viewer.vw_foo;
DROP TABLE viewer.foo CASCADE;
DROP SCHEMA viewer;
SELECT c.drop_tags, c.set_name, ddl_sql_raw, ddl_sql_sent
FROM pgl_ddl_deploy.events e
INNER JOIN pgl_ddl_deploy.set_configs c ON c.id = e.set_config_id
WHERE c.set_name = 'test1'
ORDER BY e.id DESC LIMIT 4;
                        drop_tags                         | set_name |          ddl_sql_raw           |          ddl_sql_sent          
----------------------------------------------------------+----------+--------------------------------+--------------------------------
 {"DROP SCHEMA","DROP TABLE","DROP TYPE","DROP SEQUENCE"} | test1    | DROP SCHEMA viewer;            | DROP SCHEMA viewer;
 {"DROP SCHEMA","DROP TABLE","DROP TYPE","DROP SEQUENCE"} | test1    | DROP TABLE viewer.foo CASCADE; | DROP TABLE viewer.foo CASCADE;
 {"DROP VIEW","DROP FUNCTION"}                            | test1    | DROP VIEW viewer.vw_foo;       | DROP VIEW viewer.vw_foo;
 {"DROP VIEW","DROP FUNCTION"}                            | test1    | CREATE VIEW viewer.vw_foo AS  +| CREATE VIEW viewer.vw_foo AS  +
                                                          |          | SELECT * FROM viewer.foo;      | SELECT * FROM viewer.foo;
(4 rows)

SELECT * FROM pgl_ddl_deploy.exceptions;
 id | set_name | pid | executed_at | ddl_sql | err_msg | err_state | set_config_id | resolved | resolved_notes 
----+----------+-----+-------------+---------+---------+-----------+---------------+----------+----------------
(0 rows)

DO $$
DECLARE v_ct INT;
BEGIN

IF current_setting('server_version_num')::INT >= 100000 THEN
    SELECT COUNT(1) INTO v_ct FROM pg_publication_tables WHERE schemaname = 'pgl_ddl_deploy' AND tablename = 'queue';
    RAISE LOG 'v_ct: %', v_ct;
    PERFORM verify_count(v_ct, 8);
END IF;

END$$;
